src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -758,8 +758,8 @@ public abstract class GenerateProtoTask extends DefaultTask {
       // Rewrite the trampoline file unconditionally (even if it already exists) in case the dependency or versioning
       // changes we don't need to detect the delta (and the file content is cheap to re-generate).
       String trampoline = isWindows ?
-              "@ECHO OFF\r\n\"${escapePathWindows(javaExe)}\" -jar \"${escapePathWindows(jarAbsolutePath)}\" %*\r\n" :
-              "#!/bin/sh\nexec '${escapePathUnix(javaExe)}' -jar '${escapePathUnix(jarAbsolutePath)}' \"\$@\"\n"
+              "@ECHO OFF\r\n\"${escapePathWindows(javaExe)}\" ${mainClass ? "-cp" : "-jar"} \"${escapePathWindows(jarAbsolutePath)}\" ${mainClass} %*\r\n" :
```

Comment:
Surround mainClass in quotes, and use the escaping function like used with the paths.

Reply:
I have made modifications

Reply:
You are missing the escaping; escapePathWindows() looks to be enough.